test/utils/Partition_One_Test.cpp

```diff
@@ -0,0 +1,165 @@
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include <stdio.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+#include <vector>
+
+#ifdef NGEN_WITH_SQLITE3
+#include <geopackage.hpp>
+#endif
+
+//This way we can test the partition, since this doesn't have an explicit MPI dependency
+#define NGEN_MPI_ACTIVE
+#include "core/Partition_One.hpp"
+#include "FileChecker.h"
+
+
+class PartitionOneTest: public ::testing::Test {
+
+    protected:
+
+    std::vector<std::string> hydro_fabric_paths;
+
+    std::string catchmentDataFile;
+    geojson::GeoJSON catchment_collection;
+
+    std::vector<std::string> catchment_subset_ids;
+    std::vector<std::string> nexus_subset_ids;
+
+    std::unordered_set<std::string> catchment_ids;
+    std::unordered_set<std::string> nexus_ids;
+
+    Partition_One partition_one;
+    PartitionData partition_data;
+
+    PartitionOneTest() {} 
+
+    ~PartitionOneTest() override {}
+
+    std::string file_search(const std::vector<std::string> &parent_dir_options, const std::string& file_basename)
+    {
+        // Build vector of names by building combinations of the path and basename options
+        std::vector<std::string> name_combinations;
+
+        // Build so that all path names are tried for given basename before trying a different basename option
+        for (auto & path_option : parent_dir_options)
+            name_combinations.push_back(path_option + file_basename);
+
+        return utils::FileChecker::find_first_readable(name_combinations);
+    }
+
+    void read_file_generate_partition_data()
+    {
+        const std::string file_path = file_search(hydro_fabric_paths, "catchment_data.geojson");
+        if (boost::algorithm::ends_with(catchmentDataFile, "gpkg")) {
```

Comment:
Nothing ever sets `catchmentDataFile`, so this condition will never be true. Why does the test depend on Geopackage at all? If it's not being used, just delete it!

Reply:
Agree, If we need to use gpkg in the future, we can add it back with some revision.